core/chaincode/lifecycle/deployedcc_infoprovider.go

```diff
@@ -103,6 +103,39 @@ func (vc *ValidatorCommitter) ChaincodeInfo(channelName, chaincodeName string, q
 	}, nil
 }
 
+// AllChaincodesInfo returns a list of deployed chaincodes info, including both new lifecycle and legacy chaincodes
+func (vc *ValidatorCommitter) AllChaincodesInfo(channelName string, sqe ledger.SimpleQueryExecutor) ([]*ledger.DeployedChaincodeInfo, error) {
```

Comment:
IMO, at the level of `DeployedCCInfoProvider`, it seems good to return `DeployedChaincodeInfo` to be consistent with other functions.

Reply:
As @manish-sethi said, the method returns DeployedChaincodeInfo to be consist with other functions. Plus ledger.DeployedChaincodeInfo is already used by ledger component.

Reply:
DeployedChaincodeInfoProvider returns `[]string`, `ChaincodeLifecycleInfo`, `DeployedChaincodeInfo`, `peer.StaticCollectionConfig`, `peer.CollectionConfigPackage`. Some of these return types are defined in the ledger_interface.go. Hence, returning a new type called `DeployedChaincodesNamespaces` that is `type DeployedChaincodesNamespaces map[string]string` (to be defined in the ledger_interface.go) or some better name simplifies the whole operation and then we can remove this API later once we are done with our use-case. 

As I mentioned in the initial comment, I am also okay with retaining the current function as it can be reused by other components in the future.